File: mobile_ui/quickchat_ui/lib/features/user/usecase/SearchUser.usecase.dart
import 'package:injectable/injectable.dart';
import 'package:quickchat_ui/features/authentication/data/dto/SearchUser.dto.dart';
import 'package:quickchat_ui/features/user/data/bloc/user_bloc.dart';

@injectable
class SearchUserUsecase {
  SearchUserUsecase({required this.userBloc});
  final UserBloc userBloc;

  void execute({
    required SearchUserDto searchUserDto,
  }) {
    userBloc.add(UserEvent.search(
      searchUserDto: searchUserDto,
    ));
  }
}

File: web_ui/app/features/user/domain/usecases/searchUser.usecase.ts
import { SearchUserDto } from "../../data/dto/SearchUser.dto";
import { search } from "../../data/redux/userSlice";

export class SearchUserUsecase {
  async execute(
    {
      searchUserDto,
    }: {
      searchUserDto: SearchUserDto;
    },
    dispatch: any
  ): Promise<void> {
    dispatch(search({ searchUserDto }));
  }
}
